### PR TITLE
Remove references to outputs in contrastive model.

### PR DESCRIPTION
**Pull request by owenvallis**
We use the inputs and outputs to support saving the contrastive model using the Keras API, however, we override train and test steps as well as predict. This means we don't currently support multiple output heads on the embedding output. This PR removes all references to multi-headed outputs and explicitly sets the indexer to use the predictor output.

### CODE REVIEW

File: tensorflow_similarity/models/contrastive_model.py [hunk 1]
```diff
@@ -51,25 +52,90 @@
...
+    x = inputs
+
+    for i in range(num_layers - 1):
+        x = tf.keras.layers.Dense(
+            dim,
+            use_bias=False,
+            kernel_initializer=tf.keras.initializers.LecunUniform(),
+            name=f"projector_layer_{i}",
+        )(x)
+        x = tf.keras.layers.BatchNormalization(epsilon=1.001e-5, name=f"batch_normalization_{i}")(x)
```
Comment (dnazila) [inline tensorflow_similarity/models/contrastive_model.py:66]: is epsilon=1.001e-5 well-established? should it be a constant somewhere to be used by all batchnorm layers?

File: tensorflow_similarity/models/contrastive_model.py [hunk 2]
```diff
@@ -51,25 +52,90 @@
...
+    x = tf.keras.layers.BatchNormalization(
+        epsilon=1.001e-5,
+        center=False,  # Page:5, Paragraph:2 of SimSiam paper
+        scale=False,  # Page:5, Paragraph:2 of SimSiam paper
+        name="batch_normalization_ouput",
+    )(x)
+    # Metric Logging layer. Monitors the std of the layer activations.
+    # Degnerate solutions colapse to 0 while valid solutions will move
+    # towards something like 0.0220. The actual number will depend on the layer size.
+    o = ActivationStdLoggingLayer(name="proj_std")(x)
```
Comment (dnazila) [inline tensorflow_similarity/models/contrastive_model.py:83]: better naming of the variable o

**Review by dnazila — APPROVED** [review]

File: tensorflow_similarity/models/contrastive_model.py (hunk 2 above)
Comment (owenvallis) [reply tensorflow_similarity/models/contrastive_model.py:83]: Done

File: tensorflow_similarity/models/contrastive_model.py (hunk 1 above)
Comment (owenvallis) [reply tensorflow_similarity/models/contrastive_model.py:66]: This is used to be as close as possible to the values used by the paper. I'll add it as a constant to make it consistent everywhere.
